quickwit/quickwit-proto/protos/quickwit/control_plane.proto

```diff
@@ -80,13 +81,27 @@ message ClosedShards {
   repeated uint64 shard_ids = 3;
 }
 
-// TODO: Handle partial failures.
 message GetOrCreateOpenShardsResponse {
-  repeated GetOpenShardsSubresponse subresponses = 1;
+  repeated GetOrCreateOpenShardsSuccess successes = 1;
+  repeated GetOrCreateOpenShardsFailure failures = 2;
 }
 
-message GetOpenShardsSubresponse {
-    string index_uid = 1;
-    string source_id = 2;
-    repeated quickwit.ingest.Shard open_shards = 3;
+message GetOrCreateOpenShardsSuccess {
+  uint32 subrequest_id = 1;
+  string index_uid = 2;
+  string source_id = 3;
+  repeated quickwit.ingest.Shard open_shards = 4;
+}
+
+enum GetOrCreateOpenShardsFailureReason {
+  GET_OR_CREATE_OPEN_SHARDS_FAILURE_REASON_UNSPECIFIED = 0;
```

Comment:
Hmm. Repeating the prefix might not be useful here? 
When dealing with enum in rust, we rarely `use GetOrCreateOpneShardsFailureReason::*;`
What do you think?

Reply:
on further thoughts that's specific to rust and your naming is the right one.

Reply:
Discussed offline.

The [best practice for Protobuf](https://protobuf.dev/programming-guides/dos-donts/#unspecified-enum) enums is to reuse the name of the enum as a prefix for the variant and always have `FOO_UNSPECIFIED` for the `0` tag.

Prost does the right when code-generating the corresponding Rust enums and strips the common prefix.